### ISSUE TITLE
Symfony\Component\RateLimiter\Policy\SlidingWindow::getExpirationTime(): Return value must be of type int, float returned

### ISSUE THREAD

**Issue by georgringer** [symfony/symfony#45924]
### Symfony version(s) affected

5.4.7

### Description

With the commit of https://github.com/symfony/rate-limiter/commit/4716500ba2736c9ad3b2167b32173411155be303 the method `getExpirationTime` changed and now a float is returned but the return type is set to `int` which produces an exception

### How to reproduce

No out of box code available but using `microtime(true)` returns a float

### Possible Solution

type cast like 
```
return (int)($this->windowEndAt + $this->intervalInSeconds - microtime(true));
```

**Comment by nicolas-grekas** [comment]
Cast is needed indeed. Can you please send a PR to fix this, branch 5.4, with a test case?

**Comment by georgringer** [comment]
@nicolas-grekas PR added

**Comment by georgringer** [comment]
@nicolas-grekas thanks for the merge! Can you give details for a new release? I am asking because without updating typo3 projects fail hard currently

**Comment by nicolas-grekas** [comment]
Releases happen every month, so end of April now... But this shouldn't be a fatal error. On my side, I only get a deprecation.

**Comment by georgringer** [comment]
that is kind of a pitty. in TYPO3 core we are getting exceptions with PHP8.1, e.g. in the nightly builds https://git.typo3.org/typo3/CI/cms/-/jobs/1174281

**Comment by fabpot** [comment]
We can probably do some hotfix releases in the next couple of days. Let's see first if we have some other emergency fixes.

**Comment by nicolas-grekas** [comment]
I'm not sure this need an emergency fix: from what I understand, the CI is turning a deprecation into an error. That's the issue to me and the CI should be fixed to handle deprecations more seamlessly. There's always also the possibility of locking the version to the previous version while waiting for the next release.

**Comment by sbuerk** [comment]
Granted, in this case it's not a real bug .. but could have been. For example here, the code changed and now the value may be float. If the float would been needed as return value, information would have been lost - and thus a bug. Even if reported as deprecation. Here new code has been added, but with the intention that the float fraction part is not relevant and int is the return value to go.

That's one of the reason TYPO3 raised deprecation to fail hard in the pipeline. But that's up to each project itself. 

We would have preferred a quicker release, but absolutly **understandable** why the decision is here another. At least from my side. Nothing to discuss here, maybe there is another emergency fix pressing for an earlier release which includes this fix then.

Avoid deprecation and eventually related bugs with it as early and hard as possible. Internally and for reporting to 3rd party packages. Given, if TYPO3 still would have had relaxed pipelines this would not have been detected and reported so quickly.

Having a red pipeline is a issue on our side (TYPO3), and we will find a way to deal with it. Skipping these tests for the meanwhile until the relase or something like that. Relaxing it is no option - we have found and fixed to much with E_ALL and deprecation as exception to step back here.

However - big thanks for the quick response, review and merge on a Sunday in freetime, which ensures this will be included in the next release. I think we will find a way how to deal with this in the TYPO3 core until the release.
